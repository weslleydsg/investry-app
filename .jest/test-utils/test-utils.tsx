import { NavigationContainer } from '@react-navigation/native';
import {
  render,
  RenderAPI,
  RenderOptions
} from '@testing-library/react-native';
import React from 'react';
import { QueryClientProvider } from 'react-query';
import PaperProvider from '~/providers/PaperProvider';
import { queryClient } from '~/services/queryClient';

interface Props {
  children: React.ReactElement;
}

function AllTheProviders({ children }: Props) {
  return (
    <QueryClientProvider client={queryClient}>
      <NavigationContainer>
        <PaperProvider>{children}</PaperProvider>
      </NavigationContainer>
    </QueryClientProvider>
  );
}

const customRender = (
  ui: React.ReactElement,
  options?: RenderOptions,
): RenderAPI => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';
export { customRender as render };

