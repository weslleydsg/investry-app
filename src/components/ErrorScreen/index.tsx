import React, { memo } from 'react';
import { useTranslation } from 'react-i18next';
import { Button, Headline, useTheme } from 'react-native-paper';
import { SafeAreaView } from 'react-native-safe-area-context';
import styles from './styles';

interface Props {
  networkError: boolean;
  onRetryPress(): void;
}

function ErrorScreen({ networkError, onRetryPress }: Props) {
  const { t } = useTranslation();
  const theme = useTheme();
  return (
    <SafeAreaView
      style={[styles.screen, { padding: theme.spacings.large }]}
      edges={['right', 'bottom', 'left']}
    >
      <Headline style={styles.title}>
        {t(networkError ? 'error.network' : 'error.unexpected')}
      </Headline>
      <Button
        style={{ marginTop: theme.spacings.huge }}
        mode="contained"
        onPress={onRetryPress}
      >
        {t('button.retry', { ns: 'common' })}
      </Button>
    </SafeAreaView>
  );
}

export default memo(ErrorScreen);
