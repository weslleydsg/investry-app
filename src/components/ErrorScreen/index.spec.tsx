import React from 'react';
import { fireEvent, render } from 'test-utils';
import { useTranslationMock } from '~/__mocks__/react-i18next.mock';
import ErrorScreen from '.';

describe('ErrorScreen', () => {
  it('shows network error', () => {
    const { toJSON } = render(
      <ErrorScreen networkError onRetryPress={jest.fn()} />,
    );
    const tree = toJSON();
    expect(tree).toMatchSnapshot();
    expect(useTranslationMock.self).toHaveBeenCalledWith();
    expect(useTranslationMock.t).toHaveBeenCalledWith('error.network');
    expect(useTranslationMock.t).toHaveBeenCalledWith('button.retry', {
      ns: 'common',
    });
  });

  it('shows unexpected error', () => {
    const { toJSON } = render(
      <ErrorScreen networkError={false} onRetryPress={jest.fn()} />,
    );
    const tree = toJSON();
    expect(tree).toMatchSnapshot();
    expect(useTranslationMock.self).toHaveBeenCalledWith();
    expect(useTranslationMock.t).toHaveBeenCalledWith('error.unexpected');
    expect(useTranslationMock.t).toHaveBeenCalledWith('button.retry', {
      ns: 'common',
    });
  });

  it('calls onRetryPress prop on retry button press', () => {
    const mockedOnRetryPress = jest.fn();
    const { getByText } = render(
      <ErrorScreen networkError={false} onRetryPress={mockedOnRetryPress} />,
    );
    const retryButton = getByText('button.retry');
    fireEvent.press(retryButton);
    expect(mockedOnRetryPress).toHaveBeenCalledWith();
  });
});
