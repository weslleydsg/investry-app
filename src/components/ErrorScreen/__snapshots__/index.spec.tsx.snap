// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorScreen shows network error 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    edges={
      Array [
        "right",
        "bottom",
        "left",
      ]
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "padding": 16,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 24,
                "letterSpacing": 0,
                "lineHeight": 32,
                "marginVertical": 2,
              },
              Object {
                "textAlign": "center",
              },
            ],
          ],
        ]
      }
    >
      error.network
    </Text>
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#015aa5",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "marginTop": 24,
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            button.retry
          </Text>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;

exports[`ErrorScreen shows unexpected error 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    edges={
      Array [
        "right",
        "bottom",
        "left",
      ]
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "padding": 16,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 24,
                "letterSpacing": 0,
                "lineHeight": 32,
                "marginVertical": 2,
              },
              Object {
                "textAlign": "center",
              },
            ],
          ],
        ]
      }
    >
      error.unexpected
    </Text>
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#015aa5",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "marginTop": 24,
          "minWidth": 64,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            button.retry
          </Text>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
